tests/test_core.py

```diff
@@ -282,6 +287,44 @@ def test_fetch_attachment_content(self):
                 )
             )
 
+    def test_add_attachment_content(self):
+        """Tests add_attachment_content method success."""
+        # slims_api is dynamically added to slims client
+        mock_response = MagicMock()
+        mock_response.text.return_value = self.example_add_attachments_response_text
+        with patch.object(
+            self.example_client.db.slims_api,
+            "post",
+            return_value=mock_response,
+        ) as mock_post:
+            self.example_client.add_attachment_content(
+                SlimsUnit(
+                    unit_name="test",
+                    unit_pk=1,
+                ),
+                "test",
+                "some test content",
+            )
+            assert mock_post.call_count == 1
+
+    def test_add_attachment_content_no_pk(self):
+        """Tests add_attachment_content method failure due to lack of pk."""
+        # slims_api is dynamically added to slims client
+        mock_response = MagicMock()
+        mock_response.text.return_value = self.example_add_attachments_response_text
+        with patch.object(
+            self.example_client.db.slims_api,
+            "post",
+            return_value=mock_response,
+        ) as mock_post:
+            with self.assertRaises(ValueError):
+                self.example_client.add_attachment_content(
+                    SlimsBehaviorSession(),
+                    "test",
+                    "some test content",
+                )
+            assert mock_post.call_count == 0
```

Comment:
You can usually do `mock_post.assert_not_called()`

Reply:
Addressed here [ea62847](https://github.com/AllenNeuralDynamics/aind-slims-api/pull/21/commits/ea628472bfbdd08b24804ca52a59115d61481c43)